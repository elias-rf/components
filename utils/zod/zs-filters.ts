import { zd } from "./zod";

export const zsFilters = zd.array(
  zd.object({ id: zd.string(), value: zd.string() })
);
