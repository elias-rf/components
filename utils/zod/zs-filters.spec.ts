// @ts-nocheck
import { zsFilters } from "./zs-filters";

describe("zFilters", () => {
  it("pass", () => {
    expect(zsFilters.safeParse([]).success).toEqual(true);
    expect(
      zsFilters.safeParse([
        { id: "flt1", value: "v1" },
        { id: "flt2", value: "v2" },
      ]).success
    ).toEqual(true);
    expect(
      zsFilters.safeParse([{ id: "flt", value: "v1", test: 1 }]).success
    ).toEqual(true);
  });
  it("not pass", () => {
    expect(zsFilters.safeParse({ id: "flt", value: "1" }).success).toBe(false);
    expect(zsFilters.safeParse([{ id: "flt", value: 1 }]).success).toBe(false);
    expect(zsFilters.safeParse([{ id: 1, value: "1" }]).success).toBe(false);
  });
});
