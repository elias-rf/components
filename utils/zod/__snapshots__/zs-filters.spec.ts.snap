// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`zFilters > not pass 1`] = `"__vite_ssr_import_0__.zFilters is not a function"`;

exports[`zFilters > not pass 2`] = `"__vite_ssr_import_0__.zFilters is not a function"`;

exports[`zFilters > not pass 3`] = `"__vite_ssr_import_0__.zFilters is not a function"`;
